ring-core/src/ring/util/response.clj

```diff
@@ -166,21 +175,33 @@
        (integer? (:status resp))
        (map? (:headers resp))))
 
-(defn- connection-content-length [resp ^java.net.URLConnection conn]
-  (let [content-length (.getContentLength conn)]
-    (if (neg? content-length)
-      resp
-      (header resp "Content-Length" content-length))))
+(defmulti resource-info
+  "Returns information about the resource specified by url.
+
+  The return value is a map with optional values for:
+
+  * :content        - the content of the URL, suitable for use as the :body
+                      of a ring response
+  * :content-length - the length of the :content
+  * :directory?     - true if the url points to a directory
+  * :last-modified  - the Date the :content was last modified
 
-(defn- connection-last-modified [resp ^java.net.URLConnection conn]
-  (let [last-modified (.getLastModified conn)]
-    (if (zero? last-modified)
-      resp
-      (header resp "Last-Modified" (format-date (Date. last-modified))))))
+  A value in the map may be a delay around the actual value.
 
-(defn- file-url [^java.net.URL url]
-  (if (= "file" (.getProtocol url))
-    (url-as-file url)))
+  This multimethod dispatches on the protocol of the URL as a keyword,
+  and implementations are provided for :file and :jar. If you are on a
+  platform where (Class/getResource) returns URLs with a different
+  protocol, you will need to provide an implementation for that
+  protocol.
+
+  This function is used internally by url-response."
+  {:added "1.3.3"}
```

Comment:
This metadata needs to be removed, as we don't know the version this change will be included in. It's most likely going to be Ring 1.4, but I prefer to keep version decisions out of feature pull requests.